addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -2215,12 +2225,13 @@ export class PosStore extends WithLazyGetterTrap {
             return [];
         }
 
-        const excludedProductIds = [
-            this.config.tip_product_id?.product_tmpl_id?.id,
-            ...this.session._pos_special_products_ids.map(
-                (id) => this.models["product.product"].get(id)?.product_tmpl_id?.id
-            ),
-        ].filter(Boolean);
+        // _pos_special_products_ids are product.product so we need to get their product template
+        const specialProductsTemplates = this.models["product.product"]
+            .readMany(this.session._pos_special_products_ids)
+            .filter((product) => Boolean(product)) // Remove undefined values (archived products)
+            .map((product) => product.product_tmpl_id.id);
+
+        const excludedProductIds = [this.config.tip_product_id?.id, ...specialProductsTemplates];
```

Comment:
you should remove this change. It the product.template pointed by the `tip_product_id` will be accidentally hidden after this change.